app/queries/with_one_matching_value_for_each_key.rb

```diff
@@ -0,0 +1,45 @@
+class WithOneMatchingValueForEachKey
```

Comment:
I have changed this to `WithMatchingValuesForAllKeys` its a bit wordy but it's also not a super easy concept